src/completion.h

```diff
@@ -0,0 +1,66 @@
+#ifndef HLWM_COMPLETION
+#define HLWM_COMPLETION
+
+#include "arglist.h"
+#include "types.h"
+
+/** The completion object holds the state of a running
+ * command completion, that is the list of args and the index
+ * of the element in the arglist that needs to be completed.
+ *
+ * The intended use is to pass the completion state as the reference and to
+ * return possible completions via this Completion object. This is why the
+ * operator= and the copy constructor are private. It ensures that a completion
```

Comment:
I think it makes more sense to move the explanation for why they are private to their declaration. It doesn't seem relevant to the class in general (except maybe in terms of just saying that it is non-copyable).

Reply:
Alright, I've moved them.